<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="tiltedCCA">
<title>Analysis of CITE-seq • tiltedCCA</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of CITE-seq">
<meta property="og:description" content="tiltedCCA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tiltedCCA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0.002</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/tiltedCCA.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/bm.html">Analysis of CITE-seq</a>
    <a class="dropdown-item" href="../articles/simulation.html">Tilted-CCA Simulation (3 cell types)</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Analysis of CITE-seq</h1>
            
      
      
      <div class="d-none name"><code>bm.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeuratData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://linnykos.github.io/tiltedCCA/">tiltedCCA</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>We go through how to analyze single-cell multiomic data in this
vignette. We focus on the CITE-seq data, which comes directly from the
<code>SeuratData</code> package. (In fact, this dataset is showcased in
<a href="https://satijalab.org/seurat/articles/weighted_nearest_neighbor_analysis" class="external-link uri">https://satijalab.org/seurat/articles/weighted_nearest_neighbor_analysis</a>.)</p>
<div class="section level2">
<h2 id="preliminary-analysis">Preliminary analysis<a class="anchor" aria-label="anchor" href="#preliminary-analysis"></a>
</h2>
<div class="section level3">
<h3 id="installing-and-lodaing-the-data">Installing and lodaing the data<a class="anchor" aria-label="anchor" href="#installing-and-lodaing-the-data"></a>
</h3>
<p>To install the data, uncomment the
<code>SeuratData::InstallData("bmcite")</code> line.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># SeuratData::InstallData("bmcite")</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">SeuratData</span><span class="fu">::</span><span class="fu">LoadData</span><span class="op">(</span>ds <span class="op">=</span> <span class="st">"bmcite"</span><span class="op">)</span></span></code></pre></div>
<p>This load in a dataset which 30,672 cells. In the RNA modality, there
are 17,009 genes. In the ADT modality, there are 25 surface antibody
markers.</p>
</div>
<div class="section level3">
<h3 id="basic-preprocessing-of-the-data">Basic preprocessing of the data<a class="anchor" aria-label="anchor" href="#basic-preprocessing-of-the-data"></a>
</h3>
<p>We first preprocess the RNA modality. This results in 2000 highly
variable genes, and the PCA of the RNA modality is stored in the
<code>pca</code> slot.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'RNA'</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> </span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> </span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                     verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p>Next, we preprocess the ADT modality. We treat all 25 surface
antibody markers as the highly variable genes, and the PCA of the RNA
modality is stored in the <code>apca</code> slot.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'ADT'</span></span>
<span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">bm</span><span class="op">[[</span><span class="st">"ADT"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                            normalization.method <span class="op">=</span> <span class="st">'CLR'</span>, </span>
<span>                            margin <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> </span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                     npcs <span class="op">=</span> <span class="fl">18</span>, </span>
<span>                     reduction.name <span class="op">=</span> <span class="st">'apca'</span>, </span>
<span>                     verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="basic-analysis-of-the-data">Basic analysis of the data<a class="anchor" aria-label="anchor" href="#basic-analysis-of-the-data"></a>
</h3>
<p>To illustrate what WNN does (which illustrates the “union of
information”), we can also run WNN on this dataset.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindMultiModalNeighbors.html" class="external-link">FindMultiModalNeighbors</a></span><span class="op">(</span></span>
<span>  <span class="va">bm</span>, </span>
<span>  reduction.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"pca"</span>, <span class="st">"apca"</span><span class="op">)</span>, </span>
<span>  dims.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">18</span><span class="op">)</span>, </span>
<span>  modality.weight.name <span class="op">=</span> <span class="st">"RNA.weight"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Now, we are ready to visualize the data. To do this, we run three
different UMAPS, one for RNA, one for ADT, and one for the WNN (which
combines both the RNA and ADT modality). These are stored as
<code>rna.umap</code>, <code>adt.umap</code>, and <code>wnn.umap</code>
respectively.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                      reduction <span class="op">=</span> <span class="st">'pca'</span>, </span>
<span>                      dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, assay <span class="op">=</span> <span class="st">'RNA'</span>,</span>
<span>                      reduction.name <span class="op">=</span> <span class="st">'rna.umap'</span>, </span>
<span>                      reduction.key <span class="op">=</span> <span class="st">'rnaUMAP_'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                      reduction <span class="op">=</span> <span class="st">'apca'</span>, </span>
<span>                      dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">18</span>, assay <span class="op">=</span> <span class="st">'ADT'</span>,</span>
<span>                      reduction.name <span class="op">=</span> <span class="st">'adt.umap'</span>, </span>
<span>                      reduction.key <span class="op">=</span> <span class="st">'adtUMAP_'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">bm</span>,</span>
<span>                      nn.name <span class="op">=</span> <span class="st">"weighted.nn"</span>, </span>
<span>                      reduction.name <span class="op">=</span> <span class="st">"wnn.umap"</span>, </span>
<span>                      reduction.key <span class="op">=</span> <span class="st">"wnnUMAP_"</span><span class="op">)</span></span></code></pre></div>
<p>We are now ready to plot the data. We first need to set up an
appropriate color palette.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">col_palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"CD14 Mono"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">255</span>, <span class="fl">108</span>, <span class="fl">145</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD16 Mono"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">255</span>, <span class="fl">104</span>, <span class="fl">159</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD4 Memory"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">231</span>, <span class="fl">134</span>, <span class="fl">26</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD4 Naive"</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">224</span>, <span class="fl">139</span>, <span class="fl">0</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD56 bright NK"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">0.7</span>, <span class="fl">0.7</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD8 Effector_1"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">190</span>, <span class="fl">108</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD8 Effector_2"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">188</span>, <span class="fl">89</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD8 Memory_1"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">57</span>, <span class="fl">182</span>, <span class="fl">0</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD8 Memory_2"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">186</span>, <span class="fl">66</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"CD8 Naive"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">184</span>, <span class="fl">31</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"cDC2"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">255</span>, <span class="fl">98</span>, <span class="fl">188</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"gdT"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">133</span>, <span class="fl">173</span>, <span class="fl">0</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"GMP"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">255</span>, <span class="fl">97</span>, <span class="fl">201</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"HSC"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">247</span>, <span class="fl">99</span>, <span class="fl">224</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"LMPP"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">240</span>, <span class="fl">102</span>, <span class="fl">234</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"MAIT"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">149</span>, <span class="fl">169</span>, <span class="fl">0</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"Memory B"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">180</span>, <span class="fl">239</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"Naive B"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">165</span>, <span class="fl">255</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"NK"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">0.6</span>, <span class="fl">0.6</span>, <span class="fl">0.6</span><span class="op">)</span>,</span>
<span>  <span class="st">"pDC"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">255</span>, <span class="fl">101</span>, <span class="fl">174</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"Plasmablast"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">191</span>, <span class="fl">196</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"Prog_B 1"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">172</span>, <span class="fl">136</span>, <span class="fl">255</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"Prog_B 2"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">121</span>, <span class="fl">151</span>, <span class="fl">255</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"Prog_DC"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">252</span>, <span class="fl">97</span>, <span class="fl">213</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"Prog_Mk"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">231</span>, <span class="fl">107</span>, <span class="fl">243</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"Prog_RBC"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">220</span>, <span class="fl">113</span>, <span class="fl">250</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>  <span class="st">"Treg"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">243</span>, <span class="fl">123</span>, <span class="fl">89</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                      reduction <span class="op">=</span> <span class="st">"rna.umap"</span>,</span>
<span>                      group.by <span class="op">=</span> <span class="st">"celltype.l2"</span>, </span>
<span>                      cols <span class="op">=</span> <span class="va">col_palette</span>, </span>
<span>                      label <span class="op">=</span> <span class="cn">T</span>, repel <span class="op">=</span> <span class="cn">T</span>, label.size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"RNA"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                      reduction <span class="op">=</span> <span class="st">"adt.umap"</span>,</span>
<span>                      group.by <span class="op">=</span> <span class="st">"celltype.l2"</span>, </span>
<span>                      cols <span class="op">=</span> <span class="va">col_palette</span>, </span>
<span>                      label <span class="op">=</span> <span class="cn">T</span>, repel <span class="op">=</span> <span class="cn">T</span>, label.size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"ADT"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                      reduction <span class="op">=</span> <span class="st">"wnn.umap"</span>,</span>
<span>                      group.by <span class="op">=</span> <span class="st">"celltype.l2"</span>, </span>
<span>                      cols <span class="op">=</span> <span class="va">col_palette</span>, </span>
<span>                      label <span class="op">=</span> <span class="cn">T</span>, repel <span class="op">=</span> <span class="cn">T</span>, label.size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="va">p3</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="va">p3</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"WNN"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="va">p3</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p_all</span> <span class="op">&lt;-</span> <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">p_all</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="bm_wnn.png" alt="UMAPs of the RNA modality (left), ADT modality (middle), or WNN (right, which combines the information in the RNA and ADT)" width="800px"><p class="caption">
UMAPs of the RNA modality (left), ADT modality (middle), or WNN (right,
which combines the information in the RNA and ADT)
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="running-tilted-cca">Running Tilted-CCA<a class="anchor" aria-label="anchor" href="#running-tilted-cca"></a>
</h2>
<p>We are now ready to run Tilted-CCA.</p>
<div class="section level3">
<h3 id="hard-clustering-the-cells">Hard clustering the cells<a class="anchor" aria-label="anchor" href="#hard-clustering-the-cells"></a>
</h3>
<p>Based on the visualizations, we assess that there is obvious
clustering structure in the data (which we could’ve assessed if the data
did not have cell type labels). Hence, we would like to pass “hard
clusters” (one clustering for each modality) into Tilted-CCA to get more
informative common and distinct embeddings. This results in 11 clusters
for the RNA modality, and 14 communities for the ADT modality.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"RNA"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">bm</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">bm</span>, resolution <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"ADT"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">bm</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">18</span>, reduction <span class="op">=</span> <span class="st">"apca"</span><span class="op">)</span></span>
<span><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">bm</span>, resolution <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span></code></pre></div>
<p>We can plot the clusters. This is a way to assess if the clustering
from each modality is appropriate prior to using Tilted-CCA.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                     reduction <span class="op">=</span> <span class="st">"rna.umap"</span>,</span>
<span>                     group.by <span class="op">=</span> <span class="st">"RNA_snn_res.0.25"</span>, </span>
<span>                     label <span class="op">=</span> <span class="cn">TRUE</span>, label.size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"RNA clustering"</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                     reduction <span class="op">=</span> <span class="st">"adt.umap"</span>,</span>
<span>                     group.by <span class="op">=</span> <span class="st">"ADT_snn_res.0.25"</span>, </span>
<span>                     label <span class="op">=</span> <span class="cn">TRUE</span>, label.size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"ADT clustering"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p_all</span> <span class="op">&lt;-</span> <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">p_all</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="bm_clustering.png" alt="Modality-specific clustering structure" width="800px"><p class="caption">
Modality-specific clustering structure
</p>
</div>
<p>Note: These two clusters are <em>not</em> the same. The colorings in
each of the two above plots are <em>not</em> related.</p>
</div>
<div class="section level3">
<h3 id="extracting-the-relevant-matrices">Extracting the relevant matrices<a class="anchor" aria-label="anchor" href="#extracting-the-relevant-matrices"></a>
</h3>
<p>We now extract the relevant matrices for Tilted-CCA. Since Tilted-CCA
is designed not necessarily for single-cell data, we don’t pass the
Seurat object directly into Tilted-CCA. Instead, we extract the relevant
matrices, where the rows are the cells and the columns are various
features (i.e., genes and antibodies). We also remove features that has
negligible empirical standard deviation. (In this tutorial, no genes or
antibodies were removed in this step.) We end up with
<code>mat_1b</code> that has 30,672 cells and 2000 genes, and
<code>mat_2b</code> that has the same 30,672 cells and 25 surface
antibody markers.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"RNA"</span></span>
<span><span class="va">mat_1</span> <span class="op">&lt;-</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu">t</span><span class="op">(</span><span class="va">bm</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">data</span><span class="op">[</span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">bm</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">bm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"ADT"</span></span>
<span><span class="va">mat_2</span> <span class="op">&lt;-</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu">t</span><span class="op">(</span><span class="va">bm</span><span class="op">[[</span><span class="st">"ADT"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">data</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mat_1b</span> <span class="op">&lt;-</span> <span class="va">mat_1</span></span>
<span><span class="va">sd_vec</span> <span class="op">&lt;-</span> <span class="fu">sparseMatrixStats</span><span class="fu">::</span><span class="fu">colSds</span><span class="op">(</span><span class="va">mat_1b</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="va">sd_vec</span> <span class="op">&lt;=</span> <span class="fl">1e-6</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">mat_1b</span> <span class="op">&lt;-</span> <span class="va">mat_1b</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">sd_vec</span> <span class="op">&lt;=</span> <span class="fl">1e-6</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">mat_2b</span> <span class="op">&lt;-</span> <span class="va">mat_2</span></span>
<span><span class="va">sd_vec</span> <span class="op">&lt;-</span> <span class="fu">sparseMatrixStats</span><span class="fu">::</span><span class="fu">colSds</span><span class="op">(</span><span class="va">mat_2b</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="va">sd_vec</span> <span class="op">&lt;=</span> <span class="fl">1e-6</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">mat_2b</span> <span class="op">&lt;-</span> <span class="va">mat_2b</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">sd_vec</span> <span class="op">&lt;=</span> <span class="fl">1e-6</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-tilted-cca-1">Running Tilted-CCA<a class="anchor" aria-label="anchor" href="#running-tilted-cca-1"></a>
</h3>
<p>We start Tilted-CCA by first preparing all the low-dimensional
embeddings via <code><a href="../reference/create_multiSVD.html">tiltedCCA::create_multiSVD</a></code>. Most of the
boolean arguments are meant to tell our method on whether or not
features are centered and/or scaled (or alternatively, the singular
vectors are centered and/or scaled). We would recommend the boolean
parameters here for any single-cell multiomic data that measures RNA and
antibodies.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">multiSVD_obj</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/create_multiSVD.html">create_multiSVD</a></span><span class="op">(</span>mat_1 <span class="op">=</span> <span class="va">mat_1b</span>, mat_2 <span class="op">=</span> <span class="va">mat_2b</span>,</span>
<span>                                           dims_1 <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, dims_2 <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">18</span>,</span>
<span>                                           center_1 <span class="op">=</span> <span class="cn">T</span>, center_2 <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                           normalize_row <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                           normalize_singular_value <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                           recenter_1 <span class="op">=</span> <span class="cn">F</span>, recenter_2 <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                                           rescale_1 <span class="op">=</span> <span class="cn">F</span>, rescale_2 <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                                           scale_1 <span class="op">=</span> <span class="cn">T</span>, scale_2 <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                           verbose <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>We then pass in the clusterings (one for the RNA, one for the ADT) as
well as form metacells via <code><a href="../reference/form_metacells.html">tiltedCCA::form_metacells</a></code>. The
metacells help speed up all the later Tilted-CCA calculations. In
general, we recommend using 5000 metacells if the number of cells is
dramatically larger than 5000.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">multiSVD_obj</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/form_metacells.html">form_metacells</a></span><span class="op">(</span>input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>,</span>
<span>                                          large_clustering_1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">bm</span><span class="op">$</span><span class="va">RNA_snn_res.0.25</span><span class="op">)</span>, </span>
<span>                                          large_clustering_2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">bm</span><span class="op">$</span><span class="va">ADT_snn_res.0.25</span><span class="op">)</span>, </span>
<span>                                          num_metacells <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>                                          verbose <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>This next step, using <code><a href="../reference/compute_snns.html">tiltedCCA::compute_snns</a></code>, is
arguably the most subjective step of Tilted-CCA, but nonetheless can
dramatically impact the results if not chosen with care. Here:</p>
<ul>
<li>
<code>latent_k</code> dictates how many latent dimensions are
extracted from the Laplacian bases of the shared nearest neighbor
graphs</li>
<li>
<code>num_neigh</code> dictates how many nearest neighbors are used
for each cell when constructing the nearest neighbor graphs</li>
<li>
<code>bool_cosine</code> dictates if distance between cells is based
on the cosine distance (if <code>TRUE</code>, i.e., Euclidean distance
after normalizing each gene’s low-dimensional embedding to have
Euclidean length 1) or Euclidean distance (if <code>FALSE</code>)</li>
<li>
<code>bool_intersect</code> dictates if an edge is placed between
two cells only if each cell is a nearest neighbor of the other cell (if
<code>TRUE</code>) or if an edge is placed between two cells as long as
one cell is a nearest neighbor of the other cell (if
<code>FALSE</code>)</li>
<li>
<code>min_deg</code> dictates the smallest number of edges for each
cell (which is only relevant if <code>bool_intersect=TRUE</code>), and
this value should be a value smaller or equal to
<code>num_neigh</code>
</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">multiSVD_obj</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/compute_snns.html">compute_snns</a></span><span class="op">(</span>input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>,</span>
<span>                                        latent_k <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                                        num_neigh <span class="op">=</span> <span class="fl">30</span>,</span>
<span>                                        bool_cosine <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                        bool_intersect <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                                        min_deg <span class="op">=</span> <span class="fl">30</span>,</span>
<span>                                        verbose <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p>Then, we initialize the tilt for all 18 latent dimensions (i.e., the
minimum between the number of latent dimensions in <code>dims_1</code>
and <code>dims_2</code>) via <code><a href="../reference/tiltedCCA.html">tiltedCCA::tiltedCCA</a></code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">multiSVD_obj</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/tiltedCCA.html">tiltedCCA</a></span><span class="op">(</span>input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>,</span>
<span>                                     verbose <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>Finally, we fine tune the tilt of each latent dimension. This step
can take a while. On a 2020 Macbook Pro (Big Sur, with a Quad-Core Intel
Core i7), this step takes around 20 minutes.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">multiSVD_obj</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/fine_tuning.html">fine_tuning</a></span><span class="op">(</span>input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>,</span>
<span>                                       verbose <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>After running <code>tiltedCCA:fine_tuning</code>, it is recommended
to save <code>multiSVD_obj</code> since this was the most expensive
calculation (in terms of computation time). The last function call
simply takes the tilts of each latent dimension to compute the full
decomposition of each modality (i.e., cell by feature matrix), four in
total (i.e., a common and distinct for each RNA and ADT). While this
function doesn’t take too much computation time, it greatly expands the
memory size of the <code>multiSVD_obj</code> object.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">multiSVD_obj</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/tiltedCCA_decomposition.html">tiltedCCA_decomposition</a></span><span class="op">(</span>input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>,</span>
<span>                                                   verbose <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="downstream-analysis-of-tilted-cca">Downstream analysis of Tilted-CCA<a class="anchor" aria-label="anchor" href="#downstream-analysis-of-tilted-cca"></a>
</h2>
<p>Congratulations! You have computed the common and distinct embeddings
for the single-cell multiomic data. There are various downstream
diagnostics you can apply.</p>
<div class="section level3">
<h3 id="visualizing-the-common-and-distinct-embeddings">Visualizing the common and distinct embeddings<a class="anchor" aria-label="anchor" href="#visualizing-the-common-and-distinct-embeddings"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">bm</span><span class="op">[[</span><span class="st">"common_tcca"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/create_SeuratDim.html">create_SeuratDim</a></span><span class="op">(</span>input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>,</span>
<span>                                                   what <span class="op">=</span> <span class="st">"common"</span>,</span>
<span>                                                   aligned_umap_assay <span class="op">=</span> <span class="st">"rna.umap"</span>,</span>
<span>                                                   seurat_obj <span class="op">=</span> <span class="va">bm</span>,</span>
<span>                                                   seurat_assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>                                                   verbose <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">bm</span><span class="op">[[</span><span class="st">"distinct1_tcca"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/create_SeuratDim.html">create_SeuratDim</a></span><span class="op">(</span>input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>,</span>
<span>                                                      what <span class="op">=</span> <span class="st">"distinct_1"</span>,</span>
<span>                                                      aligned_umap_assay <span class="op">=</span> <span class="st">"rna.umap"</span>,</span>
<span>                                                      seurat_obj <span class="op">=</span> <span class="va">bm</span>,</span>
<span>                                                      seurat_assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>                                                      verbose <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">bm</span><span class="op">[[</span><span class="st">"distinct2_tcca"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/create_SeuratDim.html">create_SeuratDim</a></span><span class="op">(</span>input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>,</span>
<span>                                                      what <span class="op">=</span> <span class="st">"distinct_2"</span>,</span>
<span>                                                      aligned_umap_assay <span class="op">=</span> <span class="st">"rna.umap"</span>,</span>
<span>                                                      seurat_obj <span class="op">=</span> <span class="va">bm</span>,</span>
<span>                                                      seurat_assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>                                                      verbose <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>We can now plot all three embeddings. Observe how in the common
embedding, all the major cell types such as B cells (blue), T cells
(mustard, green, and gray) and Monocytes/DC (pink) are separate from one
another. This is because both the RNA and ADT modality exhibit these
large separation structures. However, because the RNA modality does not
cleanly separate the CD8 T-cells from the CD4 T-cells (which only the
ADT modality does), there is no apparent separation between these two
T-cell sub-types in the common embedding. The RNA distinct embedding
displays further separation of the Monocytes/DC, while the ADT distinct
embedding displays further separation of the CD8 T-cells from the CD4
T-cells (green and mustard respectively).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                      reduction <span class="op">=</span> <span class="st">"common_tcca"</span>,</span>
<span>                      group.by <span class="op">=</span> <span class="st">"celltype.l2"</span>, </span>
<span>                      cols <span class="op">=</span> <span class="va">col_palette</span>, </span>
<span>                      label <span class="op">=</span> <span class="cn">T</span>, repel <span class="op">=</span> <span class="cn">T</span>, label.size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Common"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                      reduction <span class="op">=</span> <span class="st">"distinct1_tcca"</span>,</span>
<span>                      group.by <span class="op">=</span> <span class="st">"celltype.l2"</span>, </span>
<span>                      cols <span class="op">=</span> <span class="va">col_palette</span>, </span>
<span>                      label <span class="op">=</span> <span class="cn">T</span>, repel <span class="op">=</span> <span class="cn">T</span>, label.size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"RNA Distinct"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">bm</span>, </span>
<span>                      reduction <span class="op">=</span> <span class="st">"distinct2_tcca"</span>,</span>
<span>                      group.by <span class="op">=</span> <span class="st">"celltype.l2"</span>, </span>
<span>                      cols <span class="op">=</span> <span class="va">col_palette</span>, </span>
<span>                      label <span class="op">=</span> <span class="cn">T</span>, repel <span class="op">=</span> <span class="cn">T</span>, label.size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="va">p3</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="va">p3</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"ADT Distinct"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="va">p3</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p_all</span> <span class="op">&lt;-</span> <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">p_all</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="bm_tiltedcca.png" alt="Common, RNA distinct, and ADT distinct embeddings" width="800px"><p class="caption">
Common, RNA distinct, and ADT distinct embeddings
</p>
</div>
</div>
<div class="section level3">
<h3 id="assessing-how-enriched-each-cell-type-is-in-the-common-or-distinct-embeddings">Assessing how enriched each cell type is in the common or distinct
embeddings<a class="anchor" aria-label="anchor" href="#assessing-how-enriched-each-cell-type-is-in-the-common-or-distinct-embeddings"></a>
</h3>
<p>Equipped with the common and distinct embeddings, we can more
formally quantify our visual interprations from above. Specifically, we
can compute the percentage of cell types near each cell. If a cell is
“enriched” in a particular embedding (for example, the RNA distinct
embedding), this means that most of the nearest neighbors of that cell
have the same cell type. We perform these calculations using the
<code><a href="../reference/postprocess_cell_enrichment.html">tiltedCCA::postprocess_cell_enrichment</a></code> function. This takes
about 5 minutes.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">membership_vec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">bm</span><span class="op">$</span><span class="va">celltype.l2</span><span class="op">)</span></span>
<span><span class="va">cell_enrichment_res</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/postprocess_cell_enrichment.html">postprocess_cell_enrichment</a></span><span class="op">(</span></span>
<span>  input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>, </span>
<span>  membership_vec <span class="op">=</span> <span class="va">membership_vec</span>, </span>
<span>  max_subsample <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  verbose <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We can then visualize these cell-type enrichments. Observe that in
the left plot, the values go from 1 to 0 (along the x-axis), while in
the right plot, the values go from 0 to 1 (along the x-axis). To read
this plot, each cell type is represented as a dot (with color in
<code>col_palette</code>). The higher a cell type is on the y-axis, the
more it means that a cell type’s separation from all other cell types is
supported by both modalities. The more a cell type is to the extremes of
the plot (i.e., a value of 1 on either of the x-axis), the more it means
there is extra information in that modality that separates that cell
type from other cell types that is not reflect in the other modality.
The y-axis value for each cell type is the same in each plot.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_cell_enrichment.html">plot_cell_enrichment</a></span><span class="op">(</span><span class="va">cell_enrichment_res</span>,</span>
<span>                                col_palette <span class="op">=</span> <span class="va">col_palette</span>,</span>
<span>                                cex_axis <span class="op">=</span> <span class="fl">1.25</span>,</span>
<span>                                cex_lab <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>                                xlab_1 <span class="op">=</span> <span class="st">"RNA Distinct"</span>,</span>
<span>                                xlab_2 <span class="op">=</span> <span class="st">"ADT Distinct"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="bm_cell_enrichment_common-vs-distinct.png" alt="Cell type enrichment of common vs. distinct" width="600px"><p class="caption">
Cell type enrichment of common vs. distinct
</p>
</div>
</div>
<div class="section level3">
<h3 id="assessing-the-alignment-of-each-gene-with-the-common-embedding">Assessing the alignment of each gene with the common embedding<a class="anchor" aria-label="anchor" href="#assessing-the-alignment-of-each-gene-with-the-common-embedding"></a>
</h3>
<p>We now strive to assess how aligned each gene is with the common
embedding, relative to how “important” the gene is in separating the
cell types apart. We run the following code to simplify number of cell
types to consider, because in what’s to come, we will be running a
differential-expression (DE) test between each pair of cell types.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">celltype_l3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">bm</span><span class="op">$</span><span class="va">celltype.l2</span><span class="op">)</span></span>
<span><span class="va">celltype_l3</span><span class="op">[</span><span class="va">celltype_l3</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NK"</span>, <span class="st">"CD56 bright NK"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"NK_all"</span></span>
<span><span class="va">celltype_l3</span><span class="op">[</span><span class="va">celltype_l3</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MAIT"</span>, <span class="st">"gdT"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"MAIT-gdT"</span></span>
<span><span class="va">celltype_l3</span><span class="op">[</span><span class="va">celltype_l3</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Memory B"</span>, <span class="st">"Naive B"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"B"</span></span>
<span><span class="va">celltype_l3</span><span class="op">[</span><span class="va">celltype_l3</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Prog_B 1"</span>, <span class="st">"Prog_B 2"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Prog_B"</span></span>
<span><span class="va">celltype_l3</span><span class="op">[</span><span class="va">celltype_l3</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Prog_Mk"</span>, <span class="st">"Plasmablast"</span>, <span class="st">"LMPP"</span>, <span class="st">"Treg"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span><span class="va">col_palette2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">col_palette</span>, </span>
<span>                  <span class="va">col_palette</span><span class="op">[</span><span class="st">"NK"</span><span class="op">]</span>, </span>
<span>                  <span class="va">col_palette</span><span class="op">[</span><span class="st">"MAIT"</span><span class="op">]</span>, </span>
<span>                  <span class="va">col_palette</span><span class="op">[</span><span class="st">"Memory B"</span><span class="op">]</span>, </span>
<span>                  <span class="va">col_palette</span><span class="op">[</span><span class="st">"Prog_B 1"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">col_palette2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">col_palette</span><span class="op">)</span>, <span class="st">"NK_all"</span>, <span class="st">"MAIT-gdT"</span>, <span class="st">"B"</span>, <span class="st">"Prog_B"</span><span class="op">)</span></span></code></pre></div>
<p>Since there are 19 different cell types, there are 171 pairwise
comparisons to make. This would be too many comparisons for a short
tutorial. Hence, if you do not currently have time to run a multi-hour
computation, please run the following line. This subsets the cell types
to be only 4.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RECOMMEND: For a short tutorial, please run this line </span></span>
<span><span class="va">keep_celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD4 Naive"</span>, <span class="st">"CD8 Naive"</span>, <span class="st">"B"</span>, <span class="st">"CD14 Mono"</span><span class="op">)</span></span>
<span><span class="va">celltype_l3</span><span class="op">[</span><span class="op">!</span><span class="va">celltype_l3</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">keep_celltypes</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span></code></pre></div>
<p>As long as <code>celltype_l3</code> only has 4 cell types, this
following calculation would take only 20 minutes (again, on a 2020
Macbook Pro, Big Sur, with a Quad-Core Intel Core i7). The result is a
vector <code>logpval_vec</code>, which contains the negative log10
p-value for each of the 2000 highly variable genes. Please see the
documentation for <code><a href="../reference/differential_expression.html">tiltedCCA::differential_expression</a></code> on how
exactly this p-value was derived.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bm</span><span class="op">$</span><span class="va">celltype.l3</span> <span class="op">&lt;-</span> <span class="va">celltype_l3</span></span>
<span><span class="va">gene_de_list</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/differential_expression.html">differential_expression</a></span><span class="op">(</span>seurat_obj <span class="op">=</span> <span class="va">bm</span>,</span>
<span>                                                   assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>                                                   idents <span class="op">=</span> <span class="st">"celltype.l3"</span>,</span>
<span>                                                   test_use <span class="op">=</span> <span class="st">"MAST"</span>,</span>
<span>                                                   slot <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">logpval_vec</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/postprocess_depvalue.html">postprocess_depvalue</a></span><span class="op">(</span>de_list <span class="op">=</span> <span class="va">gene_de_list</span>, </span>
<span>                                               maximum_output <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span></code></pre></div>
<p>The former calculation tells us how “important” a gene is in
separating the cell types apart. We now need to compute how aligned a
gene is with the common embedding. This is what we do with the
<code><a href="../reference/postprocess_modality_alignment.html">tiltedCCA::postprocess_modality_alignment</a></code> function.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rsquare_vec</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/postprocess_modality_alignment.html">postprocess_modality_alignment</a></span><span class="op">(</span>input_obj <span class="op">=</span> <span class="va">multiSVD_obj</span>,</span>
<span>                                              bool_use_denoised <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                              seurat_obj <span class="op">=</span> <span class="va">bm</span>,</span>
<span>                                              input_assay <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                              seurat_assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>                                              seurat_slot <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span></span></code></pre></div>
<p>We are almost ready to plot <code>rsquare_vec</code> against
<code>logpval_vec</code>. Before doing so, it’s informative to compute a
few more statistics. Here, <code>gene_color[ord_idx]</code> is a color
vector that will help visually inform us of which genes have a high
sequencing depth.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">col_palette_enrichment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'lightgrey'</span>, <span class="st">'blue'</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">100</span><span class="op">)</span>, <span class="st">"33"</span><span class="op">)</span></span>
<span><span class="va">gene_breaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.18</span>, length.out <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">gene_depth</span> <span class="op">&lt;-</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">bm</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">rsquare_vec</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span><span class="op">/</span><span class="va">n</span></span>
<span><span class="va">gene_depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">gene_depth</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">gene_color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gene_depth</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">col_palette_enrichment</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.min.html" class="external-link">which.min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">gene_breaks</span> <span class="op">-</span> <span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">ord_idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">gene_depth</span>, decreasing <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p>Next, we need to focus on a few “marker” genes to display in our
plots. We can use the
<code><a href="../reference/postprocess_marker_variables.html">tiltedCCA::postprocess_marker_variables</a></code> for this matter,
since our previous call to
<code><a href="../reference/differential_expression.html">tiltedCCA::differential_expression</a></code> has already computed all
the pairwise DE tests. We also grab the cell-cycling marker genes from
<code>Seurat</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/postprocess_marker_variables.html">postprocess_marker_variables</a></span><span class="op">(</span><span class="va">de_list</span><span class="op">)</span></span>
<span><span class="va">cycling_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cc.genes</span><span class="op">$</span><span class="va">s.genes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">cc.genes</span><span class="op">$</span><span class="va">s.genes</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">logpval_vec</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                   <span class="va">cc.genes</span><span class="op">$</span><span class="va">g2m.genes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">cc.genes</span><span class="op">$</span><span class="va">g2m.genes</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">logpval_vec</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>Finally, we are ready to make the alignment plots. Here, we plot 4 of
them. The first shows all the genes, where the genes are colored by the
sequencing depth. The next two plots specifically highlight the marker
genes for either B cells or CD14 Monocytes. We see that the marker genes
are 1) to the right on the x-axis (denoting their very significant
p-values, which defines what makes a marker gene) but also 2) very high
on the y-axis (denoting that these genes express a signature that is
corroborated by both the RNA and ADT modalities). The last plot shows
the cell cycling genes, which are 1) in the middle of the x-axis
(denoting that these genes don’t differentiate any pair of cell types)
and 2) in the middle of the y-axis (denoting that these genes express a
signature that is not reflected in the ADT modality).</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span><span class="op">)</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>,<span class="fl">5</span>,<span class="fl">5</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_alignment.html">plot_alignment</a></span><span class="op">(</span>rsquare_vec <span class="op">=</span> <span class="va">rsquare_vec</span><span class="op">[</span><span class="va">ord_idx</span><span class="op">]</span>,</span>
<span>               logpval_vec <span class="op">=</span> <span class="va">logpval_vec</span><span class="op">[</span><span class="va">ord_idx</span><span class="op">]</span>,</span>
<span>               main <span class="op">=</span> <span class="st">"Human BM (CITE-Seq, RNA+ADT)\nGene differentiability vs. alignment"</span>,</span>
<span>               bool_mark_ymedian <span class="op">=</span> <span class="cn">T</span>,</span>
<span>               col_gene_highlight_border <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">255</span>, <span class="fl">205</span>, <span class="fl">87</span>, <span class="fl">255</span><span class="op">*</span><span class="fl">0.5</span>, maxColorValue <span class="op">=</span> <span class="fl">255</span><span class="op">)</span>,</span>
<span>               col_points <span class="op">=</span> <span class="va">gene_color</span><span class="op">[</span><span class="va">ord_idx</span><span class="op">]</span>,</span>
<span>               mark_median_xthres <span class="op">=</span> <span class="fl">10</span>,</span>
<span>               lwd_polygon_bold <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_alignment.html">plot_alignment</a></span><span class="op">(</span>rsquare_vec <span class="op">=</span> <span class="va">rsquare_vec</span>,</span>
<span>                 logpval_vec <span class="op">=</span> <span class="va">logpval_vec</span>,</span>
<span>                 main <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Human BM (CITE-Seq, RNA+ADT)\n"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene_list</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">" genes"</span><span class="op">)</span>,</span>
<span>                 col_gene_highlight <span class="op">=</span> <span class="va">col_palette2</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene_list</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                 gene_names <span class="op">=</span> <span class="va">gene_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                 lwd_polygon_bold <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">tiltedCCA</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_alignment.html">plot_alignment</a></span><span class="op">(</span>rsquare_vec <span class="op">=</span> <span class="va">rsquare_vec</span>,</span>
<span>               logpval_vec <span class="op">=</span> <span class="va">logpval_vec</span>,</span>
<span>               main <span class="op">=</span> <span class="st">"Human BM (CITE-Seq, RNA+ADT)\nCell cycling genes"</span>,</span>
<span>               col_gene_highlight <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>               gene_names <span class="op">=</span> <span class="va">cycling_genes</span>,</span>
<span>               lwd_polygon_bold <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="bm_gene-alignment.png" alt="Gene alignment plots, for all genes, or highlight the marker genes for B cells, CD14 Monocytes, or cell cycling" width="800px"><p class="caption">
Gene alignment plots, for all genes, or highlight the marker genes for B
cells, CD14 Monocytes, or cell cycling
</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Kevin Z Lin.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
